pkg/ttnpb/field_mask_validation.go

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ttnpb
+
+var allowedFieldMaskPathsForRPC = map[string][]string{
+	// Applications:
+	"/ttn.lorawan.v3.ApplicationRegistry/Get":                 ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationRegistry/List":                ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationRegistry/Update":              ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.EntityRegistrySearch/SearchApplications": ApplicationFieldPathsNested,
+
+	// Application Webhooks:
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/Get":  ApplicationWebhookFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/List": ApplicationWebhookFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/Set":  ApplicationWebhookFieldPathsNested,
+
+	// Application Links:
+	"/ttn.lorawan.v3.As/GetLink": ApplicationLinkFieldPathsNested,
+	"/ttn.lorawan.v3.As/SetLink": ApplicationLinkFieldPathsNested,
+
+	// Clients:
+	"/ttn.lorawan.v3.ClientRegistry/Get":                 ClientFieldPathsNested,
+	"/ttn.lorawan.v3.ClientRegistry/List":                ClientFieldPathsNested,
+	"/ttn.lorawan.v3.ClientRegistry/Update":              ClientFieldPathsNested,
+	"/ttn.lorawan.v3.EntityRegistrySearch/SearchClients": ClientFieldPathsNested,
+
+	// End Devices:
+	// TODO: Restrict field paths for IS/NS/AS/JS.
```

Comment:
This should be a reference to an issue

Reply:
Only if we're merging this without defining the actual fields in this PR